core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -261,12 +262,12 @@ private CloseableIterable<ManifestEntry<F>> open(Schema projection) {
         AvroIterable<ManifestEntry<F>> reader =
             Avro.read(file)
                 .project(ManifestEntry.wrapFileSchema(Types.StructType.of(fields)))
-                .rename("manifest_entry", GenericManifestEntry.class.getName())
-                .rename("partition", PartitionData.class.getName())
-                .rename("r102", PartitionData.class.getName())
-                .rename("data_file", content.fileClass())
-                .rename("r2", content.fileClass())
-                .classLoader(GenericManifestEntry.class.getClassLoader())
+                .createResolvingReader(
```

Comment:
Optional: I am not a big fan of how Spotless formats these closures. I would consider a helper method.

Reply:
Fixed.